src/main/java/io/coala/jetbrains/settings/ProjectSettings.java

```diff
@@ -73,4 +88,41 @@ public void removeSectionFromFilter(String section) {
     public void clearSections() {
         this.sections.clear();
     }
+
+    @Nullable
+    public Path determineAndSetExecutable(String executable) throws ExecutionException, InterruptedException, IOException {
```

Comment:
Please group the statmenets and add proper spacing, one large block of code is not readable.